File: Codeforces/Kotlin/src/main/kotlin/DislikeOfThrees.kt
package problems

import kotlin.math.pow

// Problem Link: https://codeforces.com/problemset/problem/1560/A

fun main() {
    var tests = readln().toInt()

    while (tests-- > 0) {
        val k = readln().toInt()
        var numbs = 0

        for (i in 1..k.toFloat().pow(2).toInt()) {
            if (i % 3 != 0 && i % 10 != 3) numbs++
            if (numbs == k) {
                println(i)
                break
            }
        }
    }
}

File: Codeforces/C++/problemset/Dislike of Threes.cpp
// Problem Link: https://codeforces.com/problemset/problem/1560/A

#include <iostream>
using namespace std;
typedef long long ll;
int main()
{
    ll t, k;
    cin >> t;
    for(ll i = 0; i < t; i++)
    {
        ll no = 1, fin, x = 1;
        cin >> k;
        while(x <= k)
        {
            if((no % 3) != 0 && (no % 10) != 3)
            {
                fin = no;
                x++;
            }
            no++;
        }
        cout << fin << "\n";
    }
    return 0;
}
